xbmc/utils/StringUtils.cpp

```diff
@@ -591,6 +591,43 @@ std::string& StringUtils::RemoveDuplicatedSpacesAndTabs(std::string& str)
   return str;
 }
 
+bool StringUtils::IsSpecialCharacter(char c)
+{
+  const char specialChars[] = " .-_+,!'\"\t/\\*?#$%&@()[]{}";
+
+  for (char specialChar : specialChars)
+  {
+    if (c == specialChar) {
+      return true;
+    }
+  }
+  return false;
+}
+
+std::string StringUtils::ReplaceSpecialCharactersWithSpace(const std::string& str)
```

Comment:
Could this be implemented simpler using `std::replace_if`?

Reply:
It's not only replacing, but also merge all white spaces into one. I can't see how it can be simpler to do this.

Reply:
Okay, that was just a question.